### PR TITLE
Add 20-newsgroup lucene index guide

### PR DESCRIPTION
**Pull request by x65han**
- Add a README on how to obtain Lucene index for `20-newsgroup` dataset along with helper script

### CODE REVIEW

File: docs/20newsgroup.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
+# Anserini: 20-Newsgroup
```
Comment (lintool) [inline docs/20newsgroup.md:1]: rename `experiments-20newsgroups.md` for consistency with everything else in that directory?

Comment (lintool) [reply docs/20newsgroup.md:1]: "20 Newsgroups", without dash?

File: docs/20newsgroup.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+This page contains instructions for how to index the 20-Newsgroup dataset.
```
Comment (lintool) [inline docs/20newsgroup.md:3]: same

File: docs/20newsgroup.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+## Data Prep
+
+We're going to use `20-newsgroup/` as the working directory.
+First, we need to download and extract the dataset:
+
+```sh
```
Comment (lintool) [inline docs/20newsgroup.md:10]: can we download into `collections/` so we don't clutter up top-level?

File: docs/20newsgroup.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+```
+sh target/appassembler/bin/IndexCollection -collection TwentyNewsgroupsCollection \
+ -input 20-newsgroup/20news-bydate-test \
+ -index 20-newsgroup/lucene-index.20newsgroup.test.pos+docvectors+raw \
+ -generator DefaultLuceneDocumentGenerator -threads 2 \
+ -storePositions -storeDocvectors -storeRaw
+```
+
+|               | Index Duration  | # of docs |
+|---------------|-----------------|-----------|
```
Comment (lintool) [inline docs/20newsgroup.md:63]: 2nd and 3rd col right align to make pretty.

File: docs/20newsgroup.md [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+To index the test set:
+```
+sh target/appassembler/bin/IndexCollection -collection TwentyNewsgroupsCollection \
+ -input 20-newsgroup/20news-bydate-test \
+ -index 20-newsgroup/lucene-index.20newsgroup.test.pos+docvectors+raw \
+ -generator DefaultLuceneDocumentGenerator -threads 2 \
+ -storePositions -storeDocvectors -storeRaw
+```
+
```
Comment (lintool) [inline docs/20newsgroup.md:61]: we're missing a sentence here, like "these are the stats you should see..."

File: docs/20newsgroup.md (hunk 1 above)
Comment (x65han) [reply docs/20newsgroup.md:1]: So `experiments-20newsgroups.md`?

**Comment by x65han** [comment]
Resolved all. Not sure if I got everything right. Let me know, please. Thanks!

File: src/main/python/20newsgroup/prune_and_merge.py [hunk 6]
```diff
@@ -0,0 +1,89 @@
+# -*- coding: utf-8 -*-
```
Comment (lintool) [inline src/main/python/20newsgroup/prune_and_merge.py:1]: group -> groups

File: docs/experiments-20newsgroups.md [hunk 7]
```diff
@@ -0,0 +1,71 @@
+# Anserini: 20 Newsgroup
```
Comment (lintool) [inline docs/experiments-20newsgroups.md:1]: group -> groups

all instances below...

Comment (x65han) [reply docs/experiments-20newsgroups.md:1]: My bad

**Pull request merged**